### PR TITLE
Use MailKit as default mail sender

### LINKED ISSUE #3407: SendMail Error
HOST smtp.exmail.qq.com ，Enable SSL，Port 465

This server is not mail.qq.com , it's a Tencent mail server for  enterprise .

We can use the mail box on Outlook , Foxmail , windows 10 mail client and other email clients.
[https://service.exmail.qq.com/cgi-bin/help?subtype=1&&id=28&&no=1000564#2](url)

But it cann't run on OrchardCore. 

```An error occurred while sending an email: 'Syntax error, command unrecognized. The server response was: ' ```

I cann't get more infomation form this error .

### PR DESCRIPTION
**Pull request by hishamco**
Replace System.Net.Mail.SmtpClient with MailKit.Net.Smtp.SmtpClient because it is poorly designed

Fixes #3407

### CODE REVIEW

**Comment by Jetski5822** [comment]
Licence look fine, and its a well supported piece of kit. Code replacement looks good too.

Thoughts?

**Comment by hishamco** [comment]
Your thoughts @sebastienros

**Comment by Jetski5822** [comment]
I like the replacement. Are the tests broken because of this?

**Comment by hishamco** [comment]
I don't think so, it's something related to tenant based on Blog creation, but it's work fine on Travis!!

File: src/OrchardCore.Modules/OrchardCore.Email/OrchardCore.Email.csproj [hunk 1]
```diff
@@ -17,6 +17,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.Mvc" />
     <PackageReference Include="Microsoft.Extensions.DependencyInjection" />
+    <PackageReference Include="MailKit" Version="2.1.3" />
```
Comment (Jetski5822) [inline src/OrchardCore.Modules/OrchardCore.Email/OrchardCore.Email.csproj]: upgrade to latest :)

Comment (Jetski5822) [reply src/OrchardCore.Modules/OrchardCore.Email/OrchardCore.Email.csproj]: 2.1.4

Comment (Jetski5822) [reply src/OrchardCore.Modules/OrchardCore.Email/OrchardCore.Email.csproj]: Oh, this needs to be moved to the props file

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Email/OrchardCore.Email.csproj]: Sure

**Comment by Jetski5822** [comment]
TGTM

**Comment by hishamco** [comment]
Great!! waiting for @sebastienros to merge 📆 ⌚️

**Pull request merged**